src/test/e2e/global-teardown.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs');
+const path = require('path');
+
+const countsFilePath = path.join(__dirname, 'loginCounts.json');
+
+module.exports = async () => {
+  const chalk = (await import('chalk')).default; // Dynamic import
+  let totalLogins = 0;
+
+  try {
+    if (fs.existsSync(countsFilePath)) {
+      const data = fs.readFileSync(countsFilePath, 'utf8');
+      const loginCounts = JSON.parse(data);
```

Comment:
You may want a try/catch for this too incase it fails to parse? Probably wont I imagine. So up to you

Reply:
It has never yet failed but i agree with you and will fix before it's merged!